### PR TITLE
[FIXED] Adding some missing documentation on exported symbols

### LINKED ISSUE #232: Missing documentation
I am reading through the [godoc](https://godoc.org/github.com/nats-io/go-nats-streaming) and notice that there are lots of interface methods and struct fields that are undocumented. Examples:

 * https://godoc.org/github.com/nats-io/go-nats-streaming#Options doesn't document its fields.
 * https://godoc.org/github.com/nats-io/go-nats-streaming#Subscription doesn't document most of its methods. For example `MaxPending() (int, int, error)` doesn't say anything about what the returned `int`s are.

Going forward, I propose adding a linter that requires all public entities to be documented which I think should be considered best-practise for a well-written library.

### PR DESCRIPTION
**Pull request by kozlovic**
Resolves #232

[ci skip]

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: stan.go [hunk 1]
```diff
@@ -106,15 +106,42 @@ type ConnectionLostHandler func(Conn, error)
 
 // Options can be used to a create a customized connection.
 type Options struct {
-	NatsURL            string
-	NatsConn           *nats.Conn
-	ConnectTimeout     time.Duration
-	AckTimeout         time.Duration
-	DiscoverPrefix     string
+	// NatsURL is an ULR (or comma separated list of URLs) to a node or nodes
```
Comment (derekcollison) [inline stan.go]: Spelling ULR

File: stan.go [hunk 2]
```diff
@@ -106,15 +106,42 @@ type ConnectionLostHandler func(Conn, error)
...
+	// in the cluster.
+	NatsURL string
+
+	// NatsConn is an user provided low-level NATS Connection that the streaming
```
Comment (derekcollison) [inline stan.go]: is a user (I know english sucks)

File: stan.go [hunk 3]
```diff
@@ -106,15 +106,42 @@ type ConnectionLostHandler func(Conn, error)
...
+	// the cluster. If the library does not receive an ACK after this timeout,
+	// the Publish() call (or the AckHandler) will return ErrTimeout.
+	AckTimeout time.Duration
+
+	// DiscoverPrefix is the prefix connect requests are sent to for this cluster.
+	// The default is "_STAN.discover".
+	DiscoverPrefix string
+
+	// MaxPubAcksInflight specifies how many messages can be published without
+	// getting ACKs back from server before the API blocks.
```
Comment (derekcollison) [inline stan.go]: API or flow of messages?

Comment (kozlovic) [reply stan.go]: The API (the Publish() or even PublishAsync()) call will actually block

Comment (kozlovic) [reply stan.go]: so replacing with:
```
.. back from the cluster before the Publish() or PublishAsync() calls block.
```
?

**Comment by kozlovic** [comment]
@derekcollison I have updated, please have a look again. Separate commit but will squash before merging.

**Pull request merged**